presto-parser/src/main/java/io/prestosql/sql/parser/AstBuilder.java

```diff
@@ -1476,10 +1476,13 @@ public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context)
 
         boolean distinct = isDistinct(context.setQuantifier());
 
+        SqlBaseParser.NullTreatmentContext nullTreatment = context.nullTreatment();
+
         if (name.toString().equalsIgnoreCase("if")) {
             check(context.expression().size() == 2 || context.expression().size() == 3, "Invalid number of arguments for 'if' function", context);
             check(!window.isPresent(), "OVER clause not valid for 'if' function", context);
             check(!distinct, "DISTINCT not valid for 'if' function", context);
+            check(nullTreatment == null, "Null treatment clause not valid for 'if' function", context);
```

Comment:
I don't think these checks are needed anymore, since we restrict `nullTreatment` to be paired with an `over` clause in the grammar.